homeassistant/components/doorbird.py

```diff
@@ -34,11 +43,43 @@ def setup(hass, config):
     if status[0]:
         _LOGGER.info("Connected to DoorBird at %s as %s", device_ip, username)
         hass.data[DOMAIN] = device
-        return True
     elif status[1] == 401:
         _LOGGER.error("Authorization rejected by DoorBird at %s", device_ip)
         return False
     else:
         _LOGGER.error("Could not connect to DoorBird at %s: Error %s",
                       device_ip, str(status[1]))
         return False
+
+    if config[DOMAIN].get(CONF_DOORBELL_EVENTS):
+        DoorbirdRequestView.setup_notifications(hass, device)
+
+    return True
+
+
+class DoorbirdRequestView(HomeAssistantView):
+    """Provide a page for the device to call."""
+
+    @staticmethod
+    def setup_notifications(hass, device):
+        """Register the endpoint with the device."""
+        # Provide an endpoint for the device to call to trigger events
+        hass.http.register_view(DoorbirdRequestView(hass))
+
+        # This will make HA the only service that gets doorbell events
+        url = hass.config.api.base_url + API_URL + "/" + SENSOR_DOORBELL
+        device.reset_notifications()
+        device.subscribe_notification(SENSOR_DOORBELL, url)
+
+    def __init__(self, hass):
+        """Initialize the page with access to hass data."""
+        self._hass = hass
+
+    url = API_URL
+    name = API_URL[1:].replace("/", ":")
+    extra_urls = [API_URL + "/{sensor}"]
+
+    @asyncio.coroutine
+    def get(self, request, sensor):
+        self._hass.bus.fire(DOMAIN + '_' + sensor)
```

Comment:
You have access to `hass` like this:
```py
hass = request.app['hass']
```
So you don't need to pass `hass` to the view. I also suggest you refactor the view to remove the static method, and just do that logic in `setup`.

Reply:
Also please use new style string formatting instead of string concatenation.